rm/rm-policy/rm-policy-scheduler/src/main/java/org/ow2/proactive/scheduler/resourcemanager/nodesource/policy/SchedulerAwarePolicy.java

```diff
@@ -93,6 +120,32 @@ public BooleanWrapper activate() {
         return new BooleanWrapper(true);
     }
 
+    private void waitAndConnectToScheduler() throws Exception {
+        SchedulerAuthenticationInterface authentication;
+        boolean firstException = true;
+        int trialsNumber = 0;
+        while (scheduler == null && trialsNumber <= nodeSourceRecoveryTrialsNumber) {
+            trialsNumber++;
+            try {
+                authentication = SchedulerConnection.join(schedulerUrl);
+                Credentials creds = Credentials.getCredentialsBase64(credentials);
+                scheduler = authentication.login(creds);
+                Thread.sleep(nodeSourceRecoveryTimeout);
```

Comment:
In the end, the nodeSourceRecoveryTimeout is more a delay than a timeout. If I am correct, the timeout is nodeSourceRecoveryTrialsNumber * nodeSourceRecoveryTimeout. Maybe rename the nodeSourceRecoveryTimeout variable.